### ISSUE TITLE
performance issue (equal or worse performance than Octave which shouldn't be)

### ISSUE THREAD

**Issue by precisesimulation** [precisesimulation/distmesh-julia#1]

**Comment by ChrisRackauckas** [comment]
Yes, you didn't follow any of the performance guidelines. For example,

https://github.com/precisesimulation/distmesh-julia/blob/master/DistMesh.jl#L183

this is an untyped array. You should make that an array of numbers instead of just putting numbers into an `Array{Any}`. This is causing a type-instability that is disallowing the function from being inferred which is killing the performance. Of course, just that alone would destroy the performance of the whole algorithm.

https://docs.julialang.org/en/stable/manual/performance-tips/

In addition there's a lot of small array allocations which are unnecessary when you know the size. If you want to get to Fortran speeds you'll want to stack-allocate the small arrays using StaticArrays.jl. Also, you aren't using any in-place operations which of course means that your performance will be as slow as vectorized code.
